In [2]:
import sys, os, math, string, datetime as dt, time, re
import pandas as pd, numpy as np, matplotlib.pyplot as plt, sqlite3 as sql
import emosent, emoji, string

In [3]:
from nltk.tokenize import word_tokenize, sent_tokenize
from nltk.corpus import stopwords
from nltk import pos_tag
from nltk import sentiment

In [4]:
tree = os.walk("../data/raw")

In [5]:
for folders, subfolders, files in tree:
    print(folders)
    print('--')
    print(subfolders)
    print('----')
    print(files)
    print("\n")

../data/raw
--
['BullTrades', 'Learn-n', 'Legacy Trading', 'LSV', 'Misc', 'OmegaTrades', 'Pennybois', 'Phantom Trades', 'Templar Trading', 'WayTrade']
----
['.gitkeep', 'intrinsicValueCalculator.xlsx', 'text2.db', 'text3.db', 'word_test.csv', '~$intrinsicValueCalculator.xlsx']


../data/raw\BullTrades
--
[]
----
['options-chat.db', 'stocks-chat.db']


../data/raw\Learn-n
--
[]
----
['trade-chat.db']


../data/raw\Legacy Trading
--
[]
----
['general.db', 'profits.db', 'text.db', 'trading-discussion.db']


../data/raw\LSV
--
[]
----
['options.db', 'trading.db']


../data/raw\Misc
--
[]
----
['stock-chat.db', 'text.db', 'text2.db', 'text3.db', 'text4.db', 'text5.db', 'text6.db', 'trade-ideas.db', 'wetlqd-ideas.db']


../data/raw\OmegaTrades
--
[]
----
['general.db', 'potentialdupe.db', 'sentiment.db', 'tech.db', 'text.db', 'winning.db']


../data/raw\Pennybois
--
[]
----
['stock-check.db']


../data/raw\Phantom Trades
--
[]
----
['lt-investing.db', 'trading-floor.db']


../data/raw\Templa

In [6]:
def getalldata(folder):
    master_df = pd.DataFrame(columns=["id", "name", "content", "timestamp"])
    tree = os.walk(folder)
    for folders, subfolder, files in tree:
        if folders == "../data/raw":
            next
        else:
            
            for file in files: # get database file
                connection_str = folders.replace("\\", "/") +"/" + file #setup connection string
                connect = sql.connect(connection_str)
                mycur = connect.cursor()
                mycur.execute("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name") #get available tables
                available_tables = mycur.fetchall()
                server_name = folders.replace("../data/raw\\", "") #server name to append to df
                channel_name = file.replace(".db", "") # channel name to append to df
                for i, tbl in enumerate(available_tables):
                    df = pd.read_sql(f"SELECT * FROM {tbl[0]}", con=connect)
                    df["channel"] = channel_name
                    df["server"] = server_name
                    master_df = pd.concat([master_df, df], axis=0, ignore_index=True)
                connect.close()
                print(f"processed {channel_name}. Beep Bop Boop.....")
    return master_df

In [7]:
total_df = getalldata("../data/raw")

processed options-chat. Beep Bop Boop.....
processed stocks-chat. Beep Bop Boop.....
processed trade-chat. Beep Bop Boop.....
processed general. Beep Bop Boop.....
processed profits. Beep Bop Boop.....
processed text. Beep Bop Boop.....
processed trading-discussion. Beep Bop Boop.....
processed options. Beep Bop Boop.....
processed trading. Beep Bop Boop.....
processed stock-chat. Beep Bop Boop.....
processed text. Beep Bop Boop.....
processed text2. Beep Bop Boop.....
processed text3. Beep Bop Boop.....
processed text4. Beep Bop Boop.....
processed text5. Beep Bop Boop.....
processed text6. Beep Bop Boop.....
processed trade-ideas. Beep Bop Boop.....
processed wetlqd-ideas. Beep Bop Boop.....
processed general. Beep Bop Boop.....
processed potentialdupe. Beep Bop Boop.....
processed sentiment. Beep Bop Boop.....
processed tech. Beep Bop Boop.....
processed text. Beep Bop Boop.....
processed winning. Beep Bop Boop.....
processed stock-check. Beep Bop Boop.....
processed lt-investing. B

In [8]:
total_df.server.unique()

array(['BullTrades', 'Learn-n', 'Legacy Trading', 'LSV', 'Misc',
       'OmegaTrades', 'Pennybois', 'Phantom Trades', 'Templar Trading',
       'WayTrade'], dtype=object)

In [9]:
total_df.channel.unique()

array(['options-chat', 'stocks-chat', 'trade-chat', 'general', 'profits',
       'text', 'trading-discussion', 'options', 'trading', 'stock-chat',
       'text2', 'text3', 'text4', 'text5', 'text6', 'trade-ideas',
       'wetlqd-ideas', 'potentialdupe', 'sentiment', 'tech', 'winning',
       'stock-check', 'lt-investing', 'trading-floor', 'day-trading',
       'main-trading', 'option-trading', 'general-trading'], dtype=object)

In [10]:
total_df.server = total_df.server.apply(lambda x: x if x != "Misc" else "misc")

In [11]:
total_df = total_df.sort_values(["server", "channel"])

In [12]:
total_df.drop_duplicates(subset=["id", "content"])[lambda x: (x.server == "misc")].channel.value_counts()

text6    4905
text       38
Name: channel, dtype: int64

## Dealing with Duplicate Dbs in /Misc

In [13]:
# stock-chat.db = Legacy Trading/profits.db = text3.db
total_df[lambda x: x.channel == "stock-chat"].drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id          name  \
59572  187336630749757440   Cyborg#8226   
59567  187336630749757440   Cyborg#8226   
59563  290014935499538435  toan112#9456   
59559  290014935499538435  toan112#9456   
59555  719919449863356430   bernie#3240   
...                   ...           ...   
57272  700394878462328832    Jaime#0289   
57268  700394878462328832    Jaime#0289   
57265  700394878462328832    Jaime#0289   
57266  479056760255021056  .Austin#4801   
57267  695842024955314266     Cube#4200   

                                                 content  \
59572                                          Damn nice   
59567                              What’d you pay for it   
59563                                      FB looks good   
59559                                    2.55 as it said   
59555                             510 for both contracts   
...                                                  ...   
57272  AMD is wedging and this supply of confluent wi...   
57268  /NQ at a fresh 4h demands hopefully we can see...   
57265  /ES and /NQ are flying. No idea why equities a...   
57266  I’ll be posting some juicy charting and more p...   
57267                                               Same   

                              timestamp     channel server  
59572  2020-10-14T02:30:06.383000+00:00  stock-chat   misc  
59567  2020-10-14T02:30:29.588000+00:00  stock-chat   misc  
59563  2020-10-14T02:30:32.627000+00:00  stock-chat   misc  
59559  2020-10-14T02:30:42.804000+00:00  stock-chat   misc  
59555  2020-10-14T02:30:46.699000+00:00  stock-chat   misc  
...                                 ...         ...    ...  
57272  2020-12-08T15:18:52.685000+00:00  stock-chat   misc  
57268  2020-12-08T15:21:21.489000+00:00  stock-chat   misc  
57265  2020-12-08T16:03:05.514000+00:00  stock-chat   misc  
57266  2020-12-08T16:37:44.868000+00:00  stock-chat   misc  
57267  2020-12-08T16:52:40.462000+00:00  stock-chat   misc  

[486 rows x 6 columns]

In [14]:
(total_df[lambda x: (x.channel == "profits")&(x.server == "Legacy Trading")]).drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id          name  \
41234  187336630749757440   Cyborg#8226   
41229  187336630749757440   Cyborg#8226   
41225  290014935499538435  toan112#9456   
41221  290014935499538435  toan112#9456   
41217  719919449863356430   bernie#3240   
...                   ...           ...   
38934  700394878462328832    Jaime#0289   
38930  700394878462328832    Jaime#0289   
38927  700394878462328832    Jaime#0289   
38928  479056760255021056  .Austin#4801   
38929  695842024955314266     Cube#4200   

                                                 content  \
41234                                          Damn nice   
41229                              What’d you pay for it   
41225                                      FB looks good   
41221                                    2.55 as it said   
41217                             510 for both contracts   
...                                                  ...   
38934  AMD is wedging and this supply of confluent wi...   
38930  /NQ at a fresh 4h demands hopefully we can see...   
38927  /ES and /NQ are flying. No idea why equities a...   
38928  I’ll be posting some juicy charting and more p...   
38929                                               Same   

                              timestamp  channel          server  
41234  2020-10-14T02:30:06.383000+00:00  profits  Legacy Trading  
41229  2020-10-14T02:30:29.588000+00:00  profits  Legacy Trading  
41225  2020-10-14T02:30:32.627000+00:00  profits  Legacy Trading  
41221  2020-10-14T02:30:42.804000+00:00  profits  Legacy Trading  
41217  2020-10-14T02:30:46.699000+00:00  profits  Legacy Trading  
...                                 ...      ...             ...  
38934  2020-12-08T15:18:52.685000+00:00  profits  Legacy Trading  
38930  2020-12-08T15:21:21.489000+00:00  profits  Legacy Trading  
38927  2020-12-08T16:03:05.514000+00:00  profits  Legacy Trading  
38928  2020-12-08T16:37:44.868000+00:00  profits  Legacy Trading  
38929  2020-12-08T16:52:40.462000+00:00  profits  Legacy Trading  

[486 rows x 6 columns]

In [15]:
# verdict: text3.db = stock-chat = Legacy Trading/profits
total_df[lambda x: x.channel == "text3"].drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id          name  \
66444  187336630749757440   Cyborg#8226   
66439  187336630749757440   Cyborg#8226   
66435  290014935499538435  toan112#9456   
66431  290014935499538435  toan112#9456   
66427  719919449863356430   bernie#3240   
...                   ...           ...   
64144  700394878462328832    Jaime#0289   
64140  700394878462328832    Jaime#0289   
64137  700394878462328832    Jaime#0289   
64138  479056760255021056  .Austin#4801   
64139  695842024955314266     Cube#4200   

                                                 content  \
66444                                          Damn nice   
66439                              What’d you pay for it   
66435                                      FB looks good   
66431                                    2.55 as it said   
66427                             510 for both contracts   
...                                                  ...   
64144  AMD is wedging and this supply of confluent wi...   
64140  /NQ at a fresh 4h demands hopefully we can see...   
64137  /ES and /NQ are flying. No idea why equities a...   
64138  I’ll be posting some juicy charting and more p...   
64139                                               Same   

                              timestamp channel server  
66444  2020-10-14T02:30:06.383000+00:00   text3   misc  
66439  2020-10-14T02:30:29.588000+00:00   text3   misc  
66435  2020-10-14T02:30:32.627000+00:00   text3   misc  
66431  2020-10-14T02:30:42.804000+00:00   text3   misc  
66427  2020-10-14T02:30:46.699000+00:00   text3   misc  
...                                 ...     ...    ...  
64144  2020-12-08T15:18:52.685000+00:00   text3   misc  
64140  2020-12-08T15:21:21.489000+00:00   text3   misc  
64137  2020-12-08T16:03:05.514000+00:00   text3   misc  
64138  2020-12-08T16:37:44.868000+00:00   text3   misc  
64139  2020-12-08T16:52:40.462000+00:00   text3   misc  

[486 rows x 6 columns]

<h3 style="color:red">verdict: delete text3.db and stock-chat.db from /Misc/</h3>

In [16]:
# OmegaTrades/potentialdupe = misc/trade-ideas.db = misc/text5.db 
total_df[lambda x: x.channel == "potentialdupe"].drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id               name  \
159074  673279340326420549         wes10#3165   
159068  673279340326420549         wes10#3165   
159069  548320800202293258  SargeMaximus#4491   
159070  375968444803186698     kwatson20#2892   
159063  625730781683122218    TraderMatt#9564   
...                    ...                ...   
158558  451481789110681631   Zachary1022#1017   
158554  548320800202293258  SargeMaximus#4491   
158551  451481789110681631   Zachary1022#1017   
158552  451481789110681631   Zachary1022#1017   
158553  548320800202293258  SargeMaximus#4491   

                                                  content  \
159074  https://twitter.com/realdonaldtrump/status/131...   
159068  https://www.bloomberg.com/amp/news/articles/20...   
159069   So why are stocks mooning with this election bs?   
159070  > So why are stocks mooning with this election...   
159063                  Biden’s lead is good for big tech   
...                                                   ...   
158558                           holy shit read that yall   
158554       Crazy, why do you think whoever it was sold?   
158551  <@!548320800202293258> bc they think the marke...   
158552  nah but idk, they always sell before we know,t...   
158553                                            Ah true   

                               timestamp        channel       server  
159074  2020-10-06T18:54:07.769000+00:00  potentialdupe  OmegaTrades  
159068  2020-11-05T01:54:47.610000+00:00  potentialdupe  OmegaTrades  
159069  2020-11-06T03:02:40.575000+00:00  potentialdupe  OmegaTrades  
159070  2020-11-06T04:03:00.627000+00:00  potentialdupe  OmegaTrades  
159063  2020-11-06T04:03:46.984000+00:00  potentialdupe  OmegaTrades  
...                                  ...            ...          ...  
158558  2020-12-04T21:46:05.806000+00:00  potentialdupe  OmegaTrades  
158554  2020-12-05T03:22:51.893000+00:00  potentialdupe  OmegaTrades  
158551  2020-12-05T03:23:12.691000+00:00  potentialdupe  OmegaTrades  
158552  2020-12-05T03:24:23.736000+00:00  potentialdupe  OmegaTrades  
158553  2020-12-05T03:24:26.764000+00:00  potentialdupe  OmegaTrades  

[110 rows x 6 columns]

In [17]:
# the trade ideas databate contains more unique comments
total_df[lambda x: x.channel == "trade-ideas"].drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id               name  \
110841  673279340326420549         wes10#3165   
111887  673279340326420549         wes10#3165   
111878  673279340326420549         wes10#3165   
111873  673279340326420549         wes10#3165   
111868  673279340326420549         wes10#3165   
...                    ...                ...   
110325  451481789110681631   Zachary1022#1017   
110321  548320800202293258  SargeMaximus#4491   
110318  451481789110681631   Zachary1022#1017   
110319  451481789110681631   Zachary1022#1017   
110320  548320800202293258  SargeMaximus#4491   

                                                  content  \
110841  https://twitter.com/realdonaldtrump/status/131...   
111887                                         GME halted   
111878  https://www.google.com/amp/s/www.bloomberg.com...   
111873  Waymo is an autonomous driving tech company un...   
111868  https://t.co/u4ZbtW9JX5?amp=1 \n\n\nThe Compan...   
...                                                   ...   
110325                           holy shit read that yall   
110321       Crazy, why do you think whoever it was sold?   
110318  <@!548320800202293258> bc they think the marke...   
110319  nah but idk, they always sell before we know,t...   
110320                                            Ah true   

                               timestamp      channel server  
110841  2020-10-06T18:54:07.769000+00:00  trade-ideas   misc  
111887  2020-10-08T18:07:49.653000+00:00  trade-ideas   misc  
111878  2020-10-10T20:10:08.122000+00:00  trade-ideas   misc  
111873  2020-10-10T20:11:13.856000+00:00  trade-ideas   misc  
111868  2020-10-14T19:48:17.976000+00:00  trade-ideas   misc  
...                                  ...          ...    ...  
110325  2020-12-04T21:46:05.806000+00:00  trade-ideas   misc  
110321  2020-12-05T03:22:51.893000+00:00  trade-ideas   misc  
110318  2020-12-05T03:23:12.691000+00:00  trade-ideas   misc  
110319  2020-12-05T03:24:23.736000+00:00  trade-ideas   misc  
110320  2020-12-05T03:24:26.764000+00:00  trade-ideas   misc  

[222 rows x 6 columns]

In [18]:
# trade-ideas = text5.db 
total_df[lambda x: (x.channel == "text5") & (x.server == "misc")].drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id               name  \
71008  673279340326420549         wes10#3165   
72054  673279340326420549         wes10#3165   
72045  673279340326420549         wes10#3165   
72040  673279340326420549         wes10#3165   
72035  673279340326420549         wes10#3165   
...                   ...                ...   
70492  451481789110681631   Zachary1022#1017   
70488  548320800202293258  SargeMaximus#4491   
70485  451481789110681631   Zachary1022#1017   
70486  451481789110681631   Zachary1022#1017   
70487  548320800202293258  SargeMaximus#4491   

                                                 content  \
71008  https://twitter.com/realdonaldtrump/status/131...   
72054                                         GME halted   
72045  https://www.google.com/amp/s/www.bloomberg.com...   
72040  Waymo is an autonomous driving tech company un...   
72035  https://t.co/u4ZbtW9JX5?amp=1 \n\n\nThe Compan...   
...                                                  ...   
70492                           holy shit read that yall   
70488       Crazy, why do you think whoever it was sold?   
70485  <@!548320800202293258> bc they think the marke...   
70486  nah but idk, they always sell before we know,t...   
70487                                            Ah true   

                              timestamp channel server  
71008  2020-10-06T18:54:07.769000+00:00   text5   misc  
72054  2020-10-08T18:07:49.653000+00:00   text5   misc  
72045  2020-10-10T20:10:08.122000+00:00   text5   misc  
72040  2020-10-10T20:11:13.856000+00:00   text5   misc  
72035  2020-10-14T19:48:17.976000+00:00   text5   misc  
...                                 ...     ...    ...  
70492  2020-12-04T21:46:05.806000+00:00   text5   misc  
70488  2020-12-05T03:22:51.893000+00:00   text5   misc  
70485  2020-12-05T03:23:12.691000+00:00   text5   misc  
70486  2020-12-05T03:24:23.736000+00:00   text5   misc  
70487  2020-12-05T03:24:26.764000+00:00   text5   misc  

[222 rows x 6 columns]

In [19]:
# OmegaTrades/potentialdupe = partial duplicate of misc/trade-ideas.db = misc/text5.db that are partial duplicates of text.db
total_df[lambda x: (x.channel == "text") & (x.server == "misc")].drop_duplicates(subset=["id", "content"]).sort_values("timestamp")

id                     name  \
60604  673279340326420549               wes10#3165   
60598  673279340326420549               wes10#3165   
60599  548320800202293258        SargeMaximus#4491   
60600  375968444803186698           kwatson20#2892   
60593  625730781683122218          TraderMatt#9564   
...                   ...                      ...   
62112  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
62105  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
62102  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
62100  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
62101  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   

                                                 content  \
60604  https://twitter.com/realdonaldtrump/status/131...   
60598  https://www.bloomberg.com/amp/news/articles/20...   
60599   So why are stocks mooning with this election bs?   
60600  > So why are stocks mooning with this election...   
60593                  Biden’s lead is good for big tech   
...                                                  ...   
62112                                               FSLY   
62105                                               HYLN   
62102  Cheap leap that will guarantee a profit. IVR r...   
62100  Very bullish. Broke out and held it's place ab...   
62101                                    S&P 500 futures   

                              timestamp channel server  
60604  2020-10-06T18:54:07.769000+00:00    text   misc  
60598  2020-11-05T01:54:47.610000+00:00    text   misc  
60599  2020-11-06T03:02:40.575000+00:00    text   misc  
60600  2020-11-06T04:03:00.627000+00:00    text   misc  
60593  2020-11-06T04:03:46.984000+00:00    text   misc  
...                                 ...     ...    ...  
62112  2020-12-08T15:35:58.628000+00:00    text   misc  
62105  2020-12-08T15:40:38.188000+00:00    text   misc  
62102  2020-12-08T16:10:49.635000+00:00    text   misc  
62100  2020-12-08T16:24:27.489000+00:00    text   misc  
62101  2020-12-08T16:24:30.881000+00:00    text   misc  

[439 rows x 6 columns]

<h3 style="color:red">verdict: delete OmegaTrades/potentialdupe.db, Misc/text5.db and ./trade-ideas.db + rename: text.db to trade-ideas and mv to /OmegaTrades</h3>

In [20]:
# Omega Trades/winning = misc/text2.db
total_df[lambda x: (x.channel == "text2") & (x.server == "misc")].drop_duplicates(subset=["id", "content"])

id                 name  \
63613  271708801609105408        dylskrrt#0864   
63614  673279340326420549           wes10#3165   
63617  673279340326420549           wes10#3165   
63620  281194229676376065  A_Lucky_Pucker#8227   
63623  673279340326420549           wes10#3165   
...                   ...                  ...   
64107  281194229676376065  A_Lucky_Pucker#8227   
64117  281194229676376065  A_Lucky_Pucker#8227   
64122  281194229676376065  A_Lucky_Pucker#8227   
64124  281194229676376065  A_Lucky_Pucker#8227   
64134  531986968091951105           Hippo#3098   

                                                 content  \
63613                                  another good call   
63614              PLTR this morning and a DKNG daytrade   
63617                                Quick trade on DDOG   
63620  200 scalping DOCU\nLots of other trades that u...   
63623                                      Snap lottos 🙂   
...                                                  ...   
64107  250 on AMD quick scalp play this morning just ...   
64117            <@!673279340326420549> Thank you sir ^^   
64122  $60.00 profit. \nBut bad risk to reward ratio\...   
64124             Entered too early then exited to early   
64134                       I’ve been on fire this week🔥   

                              timestamp channel server  
63613  2020-12-08T15:18:53.907000+00:00   text2   misc  
63614  2020-12-08T16:35:06.936000+00:00   text2   misc  
63617  2020-12-04T16:32:40.061000+00:00   text2   misc  
63620  2020-12-04T16:11:53.140000+00:00   text2   misc  
63623  2020-12-03T16:58:01.890000+00:00   text2   misc  
...                                 ...     ...    ...  
64107  2020-10-26T13:39:52.625000+00:00   text2   misc  
64117  2020-10-21T22:38:42.152000+00:00   text2   misc  
64122  2020-10-21T15:23:17.722000+00:00   text2   misc  
64124  2020-10-21T14:58:06.478000+00:00   text2   misc  
64134  2020-10-16T13:38:35.426000+00:00   text2   misc  

[105 rows x 6 columns]

In [21]:
# Winning has a few more comments
(total_df[lambda x: (x.channel == "winning")&(x.server == "OmegaTrades")]).drop_duplicates(subset=["id", "content"])

id                 name  \
167637  271708801609105408        dylskrrt#0864   
167638  673279340326420549           wes10#3165   
167641  673279340326420549           wes10#3165   
167644  281194229676376065  A_Lucky_Pucker#8227   
167647  673279340326420549           wes10#3165   
...                    ...                  ...   
168181  673279340326420549           wes10#3165   
168183  531986968091951105           Hippo#3098   
168189  531986968091951105           Hippo#3098   
168192  281194229676376065  A_Lucky_Pucker#8227   
168199  673279340326420549           wes10#3165   

                                                  content  \
167637                                  another good call   
167638              PLTR this morning and a DKNG daytrade   
167641                                Quick trade on DDOG   
167644  200 scalping DOCU\nLots of other trades that u...   
167647                                      Snap lottos 🙂   
...                                                   ...   
168181             Let’s go hippo base hits not home runs   
168183  It’s not much but I’m starting to get the ball...   
168189                                      Apple 116-123   
168192                           512.90 x 80\n513.19 x 79   
168199                          Quick ZM scalp cashed out   

                               timestamp  channel       server  
167637  2020-12-08T15:18:53.907000+00:00  winning  OmegaTrades  
167638  2020-12-08T16:35:06.936000+00:00  winning  OmegaTrades  
167641  2020-12-04T16:32:40.061000+00:00  winning  OmegaTrades  
167644  2020-12-04T16:11:53.140000+00:00  winning  OmegaTrades  
167647  2020-12-03T16:58:01.890000+00:00  winning  OmegaTrades  
...                                  ...      ...          ...  
168181  2020-10-14T14:49:11.936000+00:00  winning  OmegaTrades  
168183  2020-10-14T14:48:51.657000+00:00  winning  OmegaTrades  
168189  2020-10-13T15:48:46.812000+00:00  winning  OmegaTrades  
168192  2020-10-13T15:47:09.237000+00:00  winning  OmegaTrades  
168199  2020-10-13T14:25:27.390000+00:00  winning  OmegaTrades  

[114 rows x 6 columns]

<h3 style="color:red">verdict: delete Misc/text2.db</h3>

In [22]:
# text.db = text4.db
total_df[lambda x: (x.channel == "text") & (x.server == "misc")].drop_duplicates(subset=["id", "content"])

id                     name  \
59577  451481789110681631         Zachary1022#1017   
59578  451481789110681631         Zachary1022#1017   
59579  548320800202293258        SargeMaximus#4491   
59580  548320800202293258        SargeMaximus#4491   
59584  451481789110681631         Zachary1022#1017   
...                   ...                      ...   
63588  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
63593  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
63598  159985870458322944                MEE6#4876   
63603  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
63608  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   

                                                 content  \
59577  <@!548320800202293258> bc they think the marke...   
59578  nah but idk, they always sell before we know,t...   
59579                                            Ah true   
59580       Crazy, why do you think whoever it was sold?   
59584                           holy shit read that yall   
...                                                  ...   
63588  AAPL: Wait for a break to either the upside or...   
63593             MSFT: Trading Sideways, wait for break   
63598  GG <@!374316215797284877>, you just advanced t...   
63603  Costco: Ascending triangle, watch for a break ...   
63608                                    Ya, gimme a sec   

                              timestamp channel server  
59577  2020-12-05T03:23:12.691000+00:00    text   misc  
59578  2020-12-05T03:24:23.736000+00:00    text   misc  
59579  2020-12-05T03:24:26.764000+00:00    text   misc  
59580  2020-12-05T03:22:51.893000+00:00    text   misc  
59584  2020-12-04T21:46:05.806000+00:00    text   misc  
...                                 ...     ...    ...  
63588  2020-11-21T22:29:21.835000+00:00    text   misc  
63593  2020-11-21T22:29:10.607000+00:00    text   misc  
63598  2020-11-21T22:29:00.484000+00:00    text   misc  
63603  2020-11-21T22:29:00.368000+00:00    text   misc  
63608  2020-11-21T22:18:43.131000+00:00    text   misc  

[439 rows x 6 columns]

In [23]:
total_df[lambda x: (x.channel == "text4") & (x.server == "misc")].drop_duplicates(subset=["id", "content"])

id                     name  \
66449  451481789110681631         Zachary1022#1017   
66450  451481789110681631         Zachary1022#1017   
66451  548320800202293258        SargeMaximus#4491   
66452  548320800202293258        SargeMaximus#4491   
66456  451481789110681631         Zachary1022#1017   
...                   ...                      ...   
70460  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
70465  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
70470  159985870458322944                MEE6#4876   
70475  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   
70480  374316215797284877  ! - ƓŎ ƦᗴŞᚡĄᙡŊ - !#4955   

                                                 content  \
66449  <@!548320800202293258> bc they think the marke...   
66450  nah but idk, they always sell before we know,t...   
66451                                            Ah true   
66452       Crazy, why do you think whoever it was sold?   
66456                           holy shit read that yall   
...                                                  ...   
70460  AAPL: Wait for a break to either the upside or...   
70465             MSFT: Trading Sideways, wait for break   
70470  GG <@!374316215797284877>, you just advanced t...   
70475  Costco: Ascending triangle, watch for a break ...   
70480                                    Ya, gimme a sec   

                              timestamp channel server  
66449  2020-12-05T03:23:12.691000+00:00   text4   misc  
66450  2020-12-05T03:24:23.736000+00:00   text4   misc  
66451  2020-12-05T03:24:26.764000+00:00   text4   misc  
66452  2020-12-05T03:22:51.893000+00:00   text4   misc  
66456  2020-12-04T21:46:05.806000+00:00   text4   misc  
...                                 ...     ...    ...  
70460  2020-11-21T22:29:21.835000+00:00   text4   misc  
70465  2020-11-21T22:29:10.607000+00:00   text4   misc  
70470  2020-11-21T22:29:00.484000+00:00   text4   misc  
70475  2020-11-21T22:29:00.368000+00:00   text4   misc  
70480  2020-11-21T22:18:43.131000+00:00   text4   misc  

[439 rows x 6 columns]

<h3 style="color:red">remove Misc/text4.db</h3>

In [24]:
# wetlqd-ideas = text6.db
total_df[lambda x: (x.channel == "text6") & (x.server == "misc")].drop_duplicates(subset=["id", "content"])

id                            name  \
72058   488359813160501251                    Merente#5844   
72059   689681229989150742                  jackaplan#7289   
72060   612337363326074936                  Trademate#2515   
72063   131952052069728258                     Teecee#3022   
72067   464449226521313292                TradingJEDI#2647   
...                    ...                             ...   
110279  445583268792893449                itsyoboyvic#7198   
110284  445583268792893449                itsyoboyvic#7198   
110289  445583268792893449                itsyoboyvic#7198   
110294  579205396166737921  beenpippinsincebeenpippin#4039   
110307  579205396166737921  beenpippinsincebeenpippin#4039   

                                                  content  \
72058   Thanks <@464449226521313292> !  I opted for th...   
72059   Barely followed the market at all today (didn’...   
72060     <@131952052069728258> has reached level **13**!   
72063                  <@!464449226521313292> thanks boss   
72067   <@!399198468159045633> BRO AWSOME SO GLAD YOU ...   
...                                                   ...   
110279                                             Heroin   
110284                                          Big crack   
110289                                              Crack   
110294  @everyone today’s total. This doesn’t calculat...   
110307                                 Play account today   

                               timestamp channel server  
72058   2020-11-30T20:48:43.035000+00:00   text6   misc  
72059   2020-11-30T20:58:32.872000+00:00   text6   misc  
72060   2020-11-30T19:55:20.913000+00:00   text6   misc  
72063   2020-11-30T19:55:20.625000+00:00   text6   misc  
72067   2020-11-30T19:44:26.516000+00:00   text6   misc  
...                                  ...     ...    ...  
110279  2019-08-02T01:14:44.367000+00:00   text6   misc  
110284  2019-08-02T01:14:39.682000+00:00   text6   misc  
110289  2019-08-02T01:14:35.798000+00:00   text6   misc  
110294  2019-08-02T01:13:47.923000+00:00   text6   misc  
110307  2019-08-02T01:10:52.637000+00:00   text6   misc  

[4905 rows x 6 columns]

In [25]:
total_df[lambda x: (x.channel == "wetlqd-ideas") & (x.server == "misc")].drop_duplicates(subset=["id", "content"])

id                            name  \
111891  488359813160501251                    Merente#5844   
111892  689681229989150742                  jackaplan#7289   
111893  612337363326074936                  Trademate#2515   
111896  131952052069728258                     Teecee#3022   
111900  464449226521313292                TradingJEDI#2647   
...                    ...                             ...   
150112  445583268792893449                itsyoboyvic#7198   
150117  445583268792893449                itsyoboyvic#7198   
150122  445583268792893449                itsyoboyvic#7198   
150127  579205396166737921  beenpippinsincebeenpippin#4039   
150140  579205396166737921  beenpippinsincebeenpippin#4039   

                                                  content  \
111891  Thanks <@464449226521313292> !  I opted for th...   
111892  Barely followed the market at all today (didn’...   
111893    <@131952052069728258> has reached level **13**!   
111896                 <@!464449226521313292> thanks boss   
111900  <@!399198468159045633> BRO AWSOME SO GLAD YOU ...   
...                                                   ...   
150112                                             Heroin   
150117                                          Big crack   
150122                                              Crack   
150127  @everyone today’s total. This doesn’t calculat...   
150140                                 Play account today   

                               timestamp       channel server  
111891  2020-11-30T20:48:43.035000+00:00  wetlqd-ideas   misc  
111892  2020-11-30T20:58:32.872000+00:00  wetlqd-ideas   misc  
111893  2020-11-30T19:55:20.913000+00:00  wetlqd-ideas   misc  
111896  2020-11-30T19:55:20.625000+00:00  wetlqd-ideas   misc  
111900  2020-11-30T19:44:26.516000+00:00  wetlqd-ideas   misc  
...                                  ...           ...    ...  
150112  2019-08-02T01:14:44.367000+00:00  wetlqd-ideas   misc  
150117  2019-08-02T01:14:39.682000+00:00  wetlqd-ideas   misc  
150122  2019-08-02T01:14:35.798000+00:00  wetlqd-ideas   misc  
150127  2019-08-02T01:13:47.923000+00:00  wetlqd-ideas   misc  
150140  2019-08-02T01:10:52.637000+00:00  wetlqd-ideas   misc  

[4905 rows x 6 columns]

<h3 style="color:red">remove Misc/text6.db</h3>

<h3>/raw/data Cleanup</h3>
<ul>
<li>rm: text6, text5, text4, text3, text2, stock-chat, trade-ideas, OmegaTrades/potentialDup</li>
<li>rename: misc/text.db -> Omega Trades/trade-ideas.db</li>
</ul>

In [34]:
total_df_no_misc = getalldata("../data/raw")

processed options-chat. Beep Bop Boop.....
processed stocks-chat. Beep Bop Boop.....
processed trade-chat. Beep Bop Boop.....
processed general. Beep Bop Boop.....
processed profits. Beep Bop Boop.....
processed text. Beep Bop Boop.....
processed trading-discussion. Beep Bop Boop.....
processed options. Beep Bop Boop.....
processed trading. Beep Bop Boop.....
processed wetlqd-ideas. Beep Bop Boop.....
processed general. Beep Bop Boop.....
processed sentiment. Beep Bop Boop.....
processed tech. Beep Bop Boop.....
processed text. Beep Bop Boop.....
processed trade-ideas. Beep Bop Boop.....
processed winning. Beep Bop Boop.....
processed stock-check. Beep Bop Boop.....
processed lt-investing. Beep Bop Boop.....
processed trading-floor. Beep Bop Boop.....
processed day-trading. Beep Bop Boop.....
processed main-trading. Beep Bop Boop.....
processed option-trading. Beep Bop Boop.....
processed general-trading. Beep Bop Boop.....


In [35]:
total_df_no_misc = total_df_no_misc.drop_duplicates(["id", "content"])

In [36]:
total_df_no_misc.channel.unique()

array(['options-chat', 'stocks-chat', 'trade-chat', 'general', 'profits',
       'text', 'trading-discussion', 'options', 'trading', 'wetlqd-ideas',
       'sentiment', 'tech', 'trade-ideas', 'winning', 'stock-check',
       'lt-investing', 'trading-floor', 'day-trading', 'main-trading',
       'option-trading', 'general-trading'], dtype=object)

In [74]:
total_df_no_misc = total_df_no_misc.sort_values("timestamp").reset_index(drop=True)

In [75]:
total_df_no_misc.to_sql("comments", con="sqlite:///../data/interim/discord/discord.db", index_label="pk")

In [80]:
total_df_no_misc.shape

(31934, 6)

<h3>Comment Cleaning</h3>
    <ul>
        <li>Starting Point: 31,934 Unique Comments</li>
        <li>In order to get these comments ready for NLP analysis, I must:</li>
        <ul>
        <li>Remove scripted comments (those automatically genreated by bots)</li>
        <li>Identify Single Words Making Up Comments That Are Key to This Analysis:</li>
            <ul>
            <li>Emojis</li>
            <li>Specific Companies and Their Tickers</li>
            <li>Stock Trading Specific Slang</li>
            <li>User Mentions</li>
            </ul>
        </ul>
    </ul>

In [90]:
def catchBot(c):
    botCom = re.compile("\shas reached level \*\*.\d+\*\*")
    bot2Com = re.compile("\<.*\>, you just advanced to")
    botreg = botCom.search(c)
    botreg1 = bot2Com.search(c.strip())
    if (botreg != None) or (botreg1 != None):
        return 1
    else:
        return 0

#### Find Bots

In [91]:
total_df_no_misc = total_df_no_misc.assign(isBot = lambda x: x.content.apply(catchBot))

In [246]:
bots = total_df_no_misc[lambda x: x.isBot == 1].id.unique()

In [257]:
bots = np.concatenate([bots, total_df_no_misc[lambda x: x.name.str.contains("Bot#")].id.unique()], axis=0)

In [258]:
total_df_no_misc.loc[(total_df_no_misc.id.isin(bots)), "isBot"] = 1

#### Find Mentions

In [344]:
def catchMentions(comment):
    userIdCatcherT = r"(<@(!)?\d+>|@everyone)"
    mention = re.findall(userIdCatcherT, comment)
    if mention != []:
        return ",".join([re.sub(r"[<>!@]", "", x[i][0]) for i, x in enumerate([mention])])
    else:
        return ""

In [345]:
total_df = total_df_no_misc.assign(mentions = lambda x: x.content.apply(catchMentions))

#### Find Emojis

In [346]:
def catchEmojis(comment):
    emoji_list = emoji.emoji_lis(comment)
    return ",".join([e["emoji"] for e in emoji_list])

In [347]:
total_df = total_df.assign(emojis = lambda x: x.content.apply(catchEmojis))

In [348]:
total_df[lambda x: x.emojis != ""].head()

id               name  \
12  408706621942792192     zeuse4848#9335   
41  570291052494323722        Moocah#1918   
80  456226577798135808  Deleted User#0000   
87  587888385197342732         ricky#8577   
97  611594621222387738     BreadGame#4233   

                                              content  \
12                                                  😃   
41  Griffin McDonald’s is all doodoo 😂😂 <@57920539...   
80                                               🔥🔥😳😳   
87                                                  😂   
97                            Good morning everyone 😃   

                           timestamp       channel server  isBot  \
12  2019-08-02T17:55:31.789000+00:00  wetlqd-ideas   Misc      0   
41  2019-08-08T17:02:22.143000+00:00  wetlqd-ideas   Misc      0   
80  2019-08-16T15:19:59.020000+00:00  wetlqd-ideas   Misc      0   
87  2019-08-16T16:37:53.126000+00:00  wetlqd-ideas   Misc      0   
97  2019-08-19T13:34:26.860000+00:00  wetlqd-ideas   Misc      0   

              mentions   emojis  
12                            😃  
41  579205396166737921      😂,😂  
80                      🔥,🔥,😳,😳  
87                            😂  
97                            😃

In [382]:
total_df.to_sql("comments", con="sqlite:///../data/interim/discord/discord.db", index_label="pk", if_exists="replace")

#### Find Companies
* find companies using name, ticker or associated words
    * This project goal is to match and process comment sentiment and match it with the particular companies mentioned.
    * Companies are signified by a variety of words including: their traded ticker abbreviation, their full company name and by the names of members of their C-Suite.
    * In order to accurately and systematically obtain these mentioned companies, I will have to create a simple list of companies based on single words (ticker and name).
    * Once I have an understanding of the most popular companies being talked about in the groups, I can create a more refined parsing function to identify words that most likely trace back to the company.
    * I want to maintain the structure of the comments throughout the analysis (maintain initial structure), so that my NLP analysis can grade sentiment based on the original comment's grammer and word usage

In [349]:
# Discords link to stock starts with a $, but can only pull out 338 comments out of 30,000+. 
total_df[lambda x: x.content.str.contains(r"\$[A-Za-z]")]

id                    name  \
246    367506726582484996                 JK#0643   
610    445583268792893449        itsyoboyvic#7198   
1372   449633258116874250          dakotared#3395   
1921   674648888279302204           GOOFYSPY#8332   
1930   674648888279302204           GOOFYSPY#8332   
...                   ...                     ...   
31855  396160546140979212             jmorri#6551   
31856  101806058359496704  Ryan (itsGoggles)#8580   
31858  396160546140979212             jmorri#6551   
31889  101806058359496704  Ryan (itsGoggles)#8580   
31916  785007528152399913            nickb34#9557   

                                                 content  \
246    Yea, literally all thanks to you guys. Just wa...   
610                                                $EXAS   
1372   quick scalp on $SCWX with Dell buying out the ...   
1921    $IRM got small position. Will hold it overnight.   
1930                                 $Dis, interesting 🤔   
...                                                  ...   
31855           $GRIL again today <@!101806058359496704>   
31856        $CANG up too <:peepoyay:735222523683405984>   
31858  $AESE for anyone that was playing it just went...   
31889  $RUN i had on a watchlist like last week but t...   
31916     $OXY broke the resistance line on Brad's chart   

                              timestamp         channel           server  \
246    2019-08-29T21:06:45.315000+00:00    wetlqd-ideas             Misc   
610    2019-10-02T19:12:03.224000+00:00    wetlqd-ideas             Misc   
1372   2019-12-23T18:48:28.342000+00:00    wetlqd-ideas             Misc   
1921   2020-02-25T20:58:08.310000+00:00    wetlqd-ideas             Misc   
1930   2020-02-26T17:04:51.441000+00:00    wetlqd-ideas             Misc   
...                                 ...             ...              ...   
31855  2020-12-08T20:13:38.815000+00:00    main-trading  Templar Trading   
31856  2020-12-08T20:13:38.984000+00:00    main-trading  Templar Trading   
31858  2020-12-08T20:13:53.474000+00:00    main-trading  Templar Trading   
31889  2020-12-08T20:20:47.630000+00:00  option-trading  Templar Trading   
31916  2020-12-08T20:26:51.453000+00:00         trading              LSV   

       isBot            mentions emojis  
246        0                             
610        0                             
1372       0                             
1921       0                             
1930       0                          🤔  
...      ...                 ...    ...  
31855      0  101806058359496704         
31856      0                             
31858      0                             
31889      0                             
31916      0                             

[338 rows x 9 columns]

In [350]:
company_regex = re.compile(r"[A-Z]{2,4}")

In [380]:
# identifying companies based on tickers. Better representative. 338 -> 6,438 
total_df[lambda x:(x.content.str.contains(company_regex))]

id              name  \
23     445583268792893449  itsyoboyvic#7198   
26     194305752939102208      D.Ninan#3776   
47     194305752939102208      D.Ninan#3776   
54     344157708523667457  dylan.patel#4692   
56     194305752939102208      D.Ninan#3776   
...                   ...               ...   
31910  428921853403463713     Baraxton#0199   
31916  785007528152399913      nickb34#9557   
31919  594593113104842795  VelcroKarma#1526   
31930  184420431242067977           Dg#8853   
31933  176526191619080192    nCurrency#5919   

                                                 content  \
23     I took stamps .... and IAC both not looking go...   
26                Sold AMD aug23rd call for a fat profit   
47     MSFT profits im still not selling it tho just ...   
54                                       $139.50 and ATH   
56                                  $139.50 and ATH? Wym   
...                                                  ...   
31910                      @everyone EDIT is popping off   
31916     $OXY broke the resistance line on Brad's chart   
31919  if PIXY runs from being included in the Benzin...   
31930  <@428921853403463713> have you done a video in...   
31933                                             CRM ??   

                              timestamp       channel   server  isBot  \
23     2019-08-08T03:31:33.793000+00:00  wetlqd-ideas     Misc      0   
26     2019-08-08T13:33:00.968000+00:00  wetlqd-ideas     Misc      0   
47     2019-08-13T14:21:58.192000+00:00  wetlqd-ideas     Misc      0   
54     2019-08-13T19:01:16.234000+00:00  wetlqd-ideas     Misc      0   
56     2019-08-13T19:05:27.213000+00:00  wetlqd-ideas     Misc      0   
...                                 ...           ...      ...    ...   
31910  2020-12-08T20:25:14.542000+00:00       trading      LSV      0   
31916  2020-12-08T20:26:51.453000+00:00       trading      LSV      0   
31919  2020-12-08T20:27:59.342000+00:00    trade-chat  Learn-n      0   
31930  2020-12-08T20:39:17.246000+00:00       trading      LSV      0   
31933  2020-12-08T20:41:38.298000+00:00       options      LSV      0   

                 mentions emojis  
23                                
26                                
47                                
54                                
56                                
...                   ...    ...  
31910            everyone         
31916                             
31919                             
31930  428921853403463713         
31933                             

[6483 rows x 9 columns]

In [383]:
# next to do:
    # gather list of companies on the four American exchanges: DOW, SP500, NYSE and NASDAQ and their tickers
    # have dictionary parser that translates company name mentions to its assocaited ticker.
    # with the name and tickers, a more thorough pulling of the appropiate companies can take place. The pulling should take place on the str.lower value of the content column.
    # company mentions will be stored just as emojis and mentions are stored for now. (as comma-separated values column)

#### Find Slang 